src/runtime/methodbinder.cs

```diff
@@ -415,11 +416,32 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw, MethodBase info, Meth
                 }
                 else
                 {
-                    op = (arrayStart == paramIndex)
-                        // map remaining Python arguments to a tuple since
-                        // the managed function accepts it - hopefully :]
-                        ? Runtime.PyTuple_GetSlice(args, arrayStart, pyArgCount)
-                        : Runtime.PyTuple_GetItem(args, paramIndex);
+                    if(arrayStart == paramIndex)
+                    {
+                        // for a params method, we may have a sequence or single/multiple items
+                        // here we look to see if the item at the paramIndex is there or not
+                        // and then if it is a sequence itself.
+                        IntPtr item = Runtime.PyTuple_GetItem(args, paramIndex);
+                        Runtime.PyErr_Clear();
+                        if(item != IntPtr.Zero && !Runtime.PyString_Check(item) && Runtime.PySequence_Check(item))
```

Comment:
Please, add a test for `void Foo(params object[] args)` called from Python like this: `Foo([], [])`. In this case `args` should become an array of two arrays. It is unclear if this situation is handled.

Reply:
Will do!

Reply:
Added test and fixed up code when it was not passing.